lib/persisted-model.js

```diff
@@ -996,12 +997,13 @@ module.exports = function(registry) {
    * to reduce the number of results returned.
    * @param  {Number}   since    Return only changes since this checkpoint.
    * @param  {Object}   filter   Include only changes that match this filter, the same as for [#persistedmodel-find](find()).
+   * @param  {String}   tenant   Include only changes for this tenant
    * @callback {Function} callback Callback function called with `(err, changes)` arguments.  Required.
    * @param {Error} err Error object; see [Error object](http://docs.strongloop.com/display/LB/Error+object).
    * @param {Array} changes An array of [Change](#change) objects.
    */
 
-  PersistedModel.changes = function(since, filter, callback) {
+  PersistedModel.changes = function(since, filter, tenant, callback) {
```

Comment:
Ok, I think this part would be difficult to implement in a mixin with the current loopback code.

Since the "tenant" property is copied from the target PersistedModel to the Change model, one should be able to express the desire to limit synchronization to a certain tenant using existing `filter` argument only.

```js
MyModel.changes(since, { where: { tenant: tenantId } }, cb);
```

Then, inside `changes` function, we should be able to detect whether `filter` contains `tenant` clause and copy it to `changeFilter`. The only change needed from LoopBack is to allow mixins to override/customize the way how `changeFilter` is built.

```js
// in loopback
PersistedModel.createChangeFilter = function(since, modelFilter) {
  return {
    where: {
      checkpoint: {gte: since},
      modelName: this.modelName,
    },
  };
};

// in your mixin
module.exports = myMixinFunction(TargetModel, config) {
  // ...

  originalFilterFactory = TargetModel.createChangeFilter;
  TargetModel.createChangeFilter = function(since, modelFilter) {
    var tenantProperty = // find it out
    var filter = originalFilterFactory.call(this, since, modelFilter);
    if (modelFilter && modelFilter.where && modelFilter.where[tenantProperty]) {
      filter.where.tenant = modelFilter.where[tenantProperty];
    }
    return filter;
  };
};
```